### PR TITLE
[Docs] Add simple descriptions of function types: job and serving

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/tutorial/03-model-serving.ipynb [hunk 1]
```diff
@@ -493,7 +494,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.9.13"
+   "version": "3.7.7"
```
Comment (liranbg) [inline docs/tutorial/03-model-serving.ipynb:497]: why?

Comment (jillnogold) [reply docs/tutorial/03-model-serving.ipynb]: fixed

File: docs/runtimes/job-function.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+(job-function)=
+# Function of type `job`
+
+The `job` type function runs the code in a Kubernetes Pod.
+
+You can create (register) a function with basic attributes such as code, requirements, image, etc. using the 
+{py:meth}`~mlrun.projects.MlrunProject.set_function` method.
+Functions can be created from a single code, notebook file, or have access to the entire project context directory. (By adding the `with_repo=True` flag, the project context is cloned into the function runtime environment.)
```
Comment (theSaarco) [inline docs/runtimes/job-function.md:8]:
```suggestion
Functions can be created from a single code file, notebook file, or have access to the entire project context directory (by adding the `with_repo=True` flag, the project context is cloned into the function runtime environment.) 
```

File: docs/runtimes/job-function.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+You can create (register) a function with basic attributes such as code, requirements, image, etc. using the 
+{py:meth}`~mlrun.projects.MlrunProject.set_function` method.
+Functions can be created from a single code, notebook file, or have access to the entire project context directory. (By adding the `with_repo=True` flag, the project context is cloned into the function runtime environment.) 
+
+Examples:
+
+
+```python
+# register a (single) python file as a function
+project.set_function('src/data_prep.py', 'data-prep', image='mlrun/mlrun', handler='prep', kind="job")
```
Comment (theSaarco) [inline docs/runtimes/job-function.md:15]:
```suggestion
project.set_function('src/data_prep.py', name='data-prep', image='mlrun/mlrun', handler='prep', kind="job")
```

Would make it clearer as to what this parameter stands for.

File: docs/runtimes/job-function.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+# register a notebook file as a function, specify custom image and extra requirements 
+project.set_function('src/mynb.ipynb', name='test-function', image="my-org/my-image",
+                      handler="run_test", requirements="requirements.txt", kind="job")
```
Comment (theSaarco) [inline docs/runtimes/job-function.md:19]:
```suggestion
                      handler="run_test", requirements=["scikit-learn"], kind="job")
```

The `requirements` parameter is problematic (see https://jira.iguazeng.com/browse/ML-3861) - until we straighten things up, I prefer using it this way, at least in examples.

File: docs/runtimes/job-function.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+# register a module.handler as a function (requires defining the default sources/work dir, if it's not root)
+project.spec.workdir = "src"
+project.set_function(name="train", handler="training.train",  image="mlrun/mlrun", kind="job", with_repo=True)
+```
+
+To run the job:
+```
+project.run_function()
```
Comment (theSaarco) [inline docs/runtimes/job-function.md:28]:
```suggestion
project.run_function("train")
```

Must provide a function name.

File: docs/runtimes/serving-function.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+(serving-function)=
+# Function of type `serving`
+
+Deploying models in MLRun uses the function type `serving`. You can create a serving function using the `set_function()` call from a notebook. 
+You can also import an existing serving function/template from the {ref}`function-hub`.
```
Comment (theSaarco) [inline docs/runtimes/serving-function.md:5]: This sentence is true for `job` functions as well (it really depends on the function you're importing). Would put it there as well.

File: docs/runtimes/serving-function.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+This example converts a notebook to a serving function, adds a model to it, and deploys it:
+
+```python
+serving = project.set_function(name="my-serving", func="my_serving.py", kind="serving", image="mlrun/mlrun", handler="handler")
```
Comment (theSaarco) [inline docs/runtimes/serving-function.md:10]:
```suggestion
serving = project.set_function(name="my-serving", func="my_serving.ipynb", kind="serving", image="mlrun/mlrun", handler="handler")
```

You say it "converts a notebook...", so the func needs to be a notebook.

**Review by theSaarco — CHANGES_REQUESTED** [review]
Provided several comments on the examples.

File: docs/runtimes/serving-function.md (hunk 6 above)
Comment (jillnogold) [reply docs/runtimes/serving-function.md:5]: @theSaarco  - updated the job page

**Pull request merged**